src/main/java/duke/Ui/Ui.java

```diff
@@ -0,0 +1,111 @@
+package duke.Ui;
+
+import duke.DukeException.DukeException;
+import duke.Task.Task;
+import duke.TaskList.TaskList;
+
+/**
+ * Collection of the Ui
+ */
+public class Ui {
+    private final String Border = "____________________________________________________________";
+
+    /**
+     * Print a chat with specified message.
+     * @param chat Message to be printed.
+     */
+    public void printChat(String chat) {
+        System.out.println(Border + "\n" + chat + "\n" + Border);
+    }
+
+    /**
+     * Print introduction.
+     */
+    public void Hello() {
+        printChat("Hello! I'm Chrainx\n" + "What can I do for you?" );
+    }
+
+    /**
+     * Print new task added to TaskList.
+     * @param task Task to be added.
+     * @param numberOfTask Number of task in the TaskList.
+     */
+    public void addTask(Task task, int numberOfTask) {
+        printChat("Got it, I have added this task:\n"
+                + task.toString() + "\n"
+                + "Now you have " + numberOfTask
+                + " tasks in the list"
+        );
+    }
+
+    /**
+     * Print delete task from TaskList.
+     * @param task Task to be deleted.
+     * @param numberOfTask Number of task in the TaskList.
+     */
+    public void deleteTask(Task task, int numberOfTask) {
+        printChat("Noted, I have removed this task:\n"
+                + task.toString() + "\n"
+                + "Now you have " + numberOfTask
+                + " tasks in the list"
+        );
+    }
+
+    /**
+     * Print task that the status is changed.
+     * @param task Target task.
+     */
+    public void markAsDone(Task task) {
+        printChat("You have marked this task as done\n" + task.toString());
+    }
+
+    /**
+     * Print task that the status is changed.
+     * @param task Target task.
+     */
+    public void markAsNotDone(Task task) {
+        printChat("You have marked this task as not done\n" + task.toString());
+    }
+
+    /**
+     * Print bye message.
+     */
+    public void bye() {
+        printChat("Bye. Hope to see you again soon!\n" + "Wish You a wonderful day");
+    }
+
+    /**
+     * Print message when there is unknown input.
+     * @param e Exception where there is unknown input.
+     */
+    public void error(DukeException e) {
+        printChat(e.toString());
+    }
+
+    /**
+     * Print all tasks as a list.
+     * @param tasks TaskList of the saved tasks.
+     */
+    public void listing(TaskList tasks) {
+        String listOfTask = "Here are the list of your task:\n";
+        for (int i = 0; i < tasks.getNumberOfTask(); i++) {
+            listOfTask = listOfTask + tasks.getTask(i).toString() + "\n";
+        }
+        printChat(listOfTask);
+    }
```

Comment:
Perhaps you may want to maintain consistency of method names using verbs and avoid gerunds (verb + ing)

So in this case, `list(TaskList tasks)` may be better than `listing(TaskList tasks)`. An alternative name could be `listAllTasks(TaskList tasks)` to make it more descriptive and easier to read.